pkg/cmd/update.go

```diff
@@ -0,0 +1,165 @@
+// Copyright 2024 Daytona Platforms Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package cmd
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/daytonaio/daytona/internal"
+	"github.com/inconshreveable/go-update"
+	"github.com/spf13/cobra"
+	"golang.org/x/mod/semver"
+	"net/http"
+	"net/url"
+	"regexp"
+	"runtime"
+	"strings"
+)
+
+const (
+	githubReleaseURL = "https://api.github.com/repos/daytonaio/daytona/releases"
+	baseDownloadURL  = "https://download.daytona.io/daytona/"
+)
+
+type GitHubRelease struct {
+	TagName   string `json:"tag_name"`
+	ChangeLog string `json:"body"`
+}
+
+var versionFlag string
+var updateCmd = &cobra.Command{
+	Use:   "update",
+	Short: "Update Daytona CLI",
+	Long:  "Update Daytona CLI to the latest version",
+	Args:  cobra.NoArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		var version string
+		var changeLog string
+		currentVersion := internal.Version
+		if versionFlag == "" {
+			release, err := fetchLatestRelase()
+			if err != nil {
+				return err
+			}
+			version = release.TagName
+			changeLog = release.ChangeLog
+		} else {
+			semverRegex := `^v?(\d+\.\d+\.\d+)$`
+			matched, err := regexp.MatchString(semverRegex, versionFlag)
+			if err != nil {
+				return err
+			}
+			if !matched {
+				return errors.New("invalid version format: expected 'vX.Y.X' or 'X.Y.Z'")
+			}
+			if !strings.HasPrefix(versionFlag, "v") {
+				versionFlag = "v" + versionFlag
+			}
+			release, err := fetchVersionRelease(versionFlag)
+			if err != nil {
+				return err
+			}
+			version = release.TagName
+			changeLog = release.ChangeLog
+		}
+		if semver.Compare(version, currentVersion) <= 0 {
+			fmt.Println("Current version is greater than the version you are trying to update to")
+			return nil
+		}
+		if semver.Compare(currentVersion, version) != 0 {
+			changeLog += "\nThere are more changes in the version. Please visit https://github.com/daytonaio/daytona/releases for all the complete chnageLog"
+		}
+		fmt.Println("Updating to version ", version, "from ", currentVersion)
```

Comment:
![Screenshot 2024-12-10 at 17 14 46](https://github.com/user-attachments/assets/196663b2-13d8-40f5-af5a-9f6cefe2bfb4)

Seems that there are some extra spaces here.